QuickDebugSourceInstallingCompilationParticipantTest is too often flaky on macOS

No matter how hard I try, this test often fails on GitHub Actions macOS, which is arm.

I tried many things, including refreshing after the build and also separating the Xtext Build from the Java build, https://github.com/LorenzoBettini/xtext/commit/4b3a8e3c5d2395cc9b371b64ec21acaac74a42bd

On macOS, it fails often.

I can only think of 2 solutions

- disable this test on macOS
- use our `@Flaky` annotation at least for this test

@szarnekow @cdietrich ?

side discussion

```
java.lang.AssertionError: Couldn't find the mandatory library 'org.eclipse.xtext.xbase.lib' 2.8.0 or higher on the project's classpath.
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.assertTrue(Assert.java:42)
	at org.junit.Assert.assertFalse(Assert.java:65)
	at org.eclipse.xtend.ide.tests.compiler.Bug473833Test.assertNoErrors(Bug473833Test.java:360)
	at org.eclipse.xtend.ide.tests.compiler.Bug473833Test.testInheritedNestedTypes_01(Bug473833Test.java:86)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:93)
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:40)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:520)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:748)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:443)
	at org.eclipse.pde.internal.junit.runtime.RemotePluginTestRunner.main(RemotePluginTestRunner.java:83)
	at org.eclipse.pde.internal.junit.runtime.PlatformUITestHarness.lambda$0(PlatformUITestHarness.java:45)
	at org.eclipse.e4.ui.internal.workbench.swt.E4Testable.lambda$1(E4Testable.java:127)
	at org.eclipse.swt.widgets.RunnableLock.run(RunnableLock.java:40)
	at org.eclipse.swt.widgets.Synchronizer.runAsyncMessages(Synchronizer.java:132)
	at org.eclipse.swt.widgets.Display.runAsyncMessages(Display.java:4382)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:4005)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1151)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:339)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1042)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:153)
	at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:668)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:339)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:576)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:173)
	at org.eclipse.ui.internal.ide.application.IDEApplication.start(IDEApplication.java:178)
	at org.eclipse.pde.internal.junit.runtime.NonUIThreadTestApplication.start(NonUIThreadTestApplication.java:58)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:208)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:149)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:115)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:467)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:298)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:668)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:605)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1481)
	at org.eclipse.equinox.launcher.Main.main(Main.java:1454)
```
this is the thing you mentioned yesterday. do we have an idea?

--- a/org.eclipse.xtend.ide.tests/src/org/eclipse/xtend/ide/tests/compiler/QuickDebugSourceInstallingCompilationParticipantTest.java
+++ b/org.eclipse.xtend.ide.tests/src/org/eclipse/xtend/ide/tests/compiler/QuickDebugSourceInstallingCompilationParticipantTest.java
@@ -69,7 +69,7 @@ public class QuickDebugSourceInstallingCompilationParticipantTest extends Abstra
 			r.accept(new ClassVisitor(Opcodes.ASM9) {
 				@Override
 				public void visitSource(final String source, final String debug) {
-					if ("Outer.java".equals(source)) {
+					if ("Outer.java".equals(source) && debug != null) {
 						assertEquals("""
 							SMAP
 							Outer.java

--- a/org.eclipse.xtend.ide.tests/src/org/eclipse/xtend/ide/tests/compiler/QuickDebugSourceInstallingCompilationParticipantTest.java
+++ b/org.eclipse.xtend.ide.tests/src/org/eclipse/xtend/ide/tests/compiler/QuickDebugSourceInstallingCompilationParticipantTest.java
@@ -29,7 +29,6 @@ import org.objectweb.asm.ClassReader;
 import org.objectweb.asm.ClassVisitor;
 import org.objectweb.asm.Opcodes;
 
-import com.google.common.io.ByteStreams;
 import com.google.inject.Inject;
 
 /**
@@ -66,8 +65,7 @@ public class QuickDebugSourceInstallingCompilationParticipantTest extends Abstra
 		final AtomicBoolean debugInfoFound = new AtomicBoolean(false);
 		Path classFile = clazz.getLocation().toPath();
 		try (var in = Files.newInputStream(classFile)) {
-			final byte[] bytes = ByteStreams.toByteArray(in);
-			final ClassReader r = new ClassReader(bytes);
+			final ClassReader r = new ClassReader(in);
 			r.accept(new ClassVisitor(Opcodes.ASM9) {
 				@Override
 				public void visitSource(final String source, final String debug) {

--- a/org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/generator/trace/TraceForStorageProvider.java
+++ b/org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/generator/trace/TraceForStorageProvider.java
@@ -68,7 +68,7 @@ public class TraceForStorageProvider extends AbstractTraceForURIProvider<IFile, 
 		}
 
 		@Override
-		public long getModificationstamp() {
+		public long getModificationStamp() {
 			return file.getModificationStamp();
 		}
 

--- a/org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/generator/trace/TraceForStorageProvider.java
+++ b/org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/generator/trace/TraceForStorageProvider.java
@@ -68,7 +68,7 @@ public class TraceForStorageProvider extends AbstractTraceForURIProvider<IFile, 
 		}
 
 		@Override
-		public long getTimestamp() {
+		public long getModificationstamp() {
 			return file.getModificationStamp();
 		}
 

--- a/org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/generator/trace/TraceForStorageProvider.java
+++ b/org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/generator/trace/TraceForStorageProvider.java
@@ -69,7 +69,7 @@ public class TraceForStorageProvider extends AbstractTraceForURIProvider<IFile, 
 
 		@Override
 		public long getTimestamp() {
-			return file.getLocalTimeStamp();
+			return file.getModificationStamp();
 		}
 
 		@Override

--- a/org.eclipse.xtext/src/org/eclipse/xtext/generator/trace/internal/AbstractTraceForURIProvider.java
+++ b/org.eclipse.xtext/src/org/eclipse/xtext/generator/trace/internal/AbstractTraceForURIProvider.java
@@ -54,7 +54,7 @@ public abstract class AbstractTraceForURIProvider<SomeFile, Trace extends Abstra
 	
 	protected interface PersistedTrace {
 		AbsoluteURI getPath();
-		long getModificationstamp();
+		long getModificationStamp();
 		InputStream openStream() throws IOException;
 		boolean exists();
 	}
@@ -69,7 +69,7 @@ public abstract class AbstractTraceForURIProvider<SomeFile, Trace extends Abstra
 
 			public CacheKey(PersistedTrace source) {
 				this.path = source.getPath();
-				this.timestamp = source.getModificationstamp();
+				this.timestamp = source.getModificationStamp();
 			}
 
 			@Override

--- a/org.eclipse.xtext/src/org/eclipse/xtext/generator/trace/internal/AbstractTraceForURIProvider.java
+++ b/org.eclipse.xtext/src/org/eclipse/xtext/generator/trace/internal/AbstractTraceForURIProvider.java
@@ -54,7 +54,7 @@ public abstract class AbstractTraceForURIProvider<SomeFile, Trace extends Abstra
 	
 	protected interface PersistedTrace {
 		AbsoluteURI getPath();
-		long getTimestamp();
+		long getModificationstamp();
 		InputStream openStream() throws IOException;
 		boolean exists();
 	}
@@ -69,7 +69,7 @@ public abstract class AbstractTraceForURIProvider<SomeFile, Trace extends Abstra
 
 			public CacheKey(PersistedTrace source) {
 				this.path = source.getPath();
-				this.timestamp = source.getTimestamp();
+				this.timestamp = source.getModificationstamp();
 			}
 
 			@Override

